include/nvexec/stream/reduce.cuh

```diff
@@ -46,8 +46,16 @@ namespace nvexec {
 
         template <class Range>
         static void set_value_impl(base::__t&& self, Range&& range) noexcept {
+          cudaError_t status{cudaSuccess};
           cudaStream_t stream = self.op_state_.get_stream();
 
+          // `range` is produced asynchronously, so we need to wait for it to be ready
+          if (status = STDEXEC_DBG_ERR(cudaStreamSynchronize(stream));
+              status != cudaSuccess) {
+            self.op_state_.propagate_completion_signal(stdexec::set_error, std::move(status));
+            return;
+          }
```

Comment:
Reduce used to access potentially asynchronous value on the host. Making `schedule_from` asynchronous revealed the race. @miscco I think we need a reduce overload that accepts a range as input rather than in the value channel to provide efficient implementation when the range is known beforehand:

```cpp
bulk(N, update(field)) | reduce(field, field + n)
```